src/group.cr

```diff
@@ -0,0 +1,184 @@
+require "c/sys/limits"
+
+class Group
+  # Converts group ID into a groupname.
+  #
+  # Returns: The groupname for the given group ID.
+  #
+  # ```
+  # Group.name(0) # => root
+  # ```
+  def self.name(gid : Int) : String
+    group_struct = LibC.getgrgid(int_to_gid(gid))
+    return String.new(group_struct.value.gr_name) if !group_struct.null?
+    raise NotFound.new("Group with gid '#{gid}', was not found.")
+  end
+
+  # Converts groupname into group ID.
+  #
+  # Returns: The group ID for the given groupname.
+  # Raises: Group::NotFound error if no group exists with the given groupname.
+  #
+  # ```
+  # Group.gid("root") # => 0
+  # ```
+  def self.gid(group : String | Symbol) : Int
+    group_struct = LibC.getgrnam(group.to_s.check_no_null_byte)
+    return group_struct.value.gr_gid if !group_struct.null?
+    raise NotFound.new("Group with name '#{group.to_s}', was not found.")
+  end
+
+  # Checks to see if a group with the given group ID exists.
+  #
+  # Returns: `true` if the group exists.
+  #
+  # ```
+  # Group.exists?("root")              # => true
+  # Group.exists?("nonexistant_group") # => false
+  # ```
+  def self.exists?(gid : Int) : Bool
+    return !LibC.getgrgid(int_to_gid(gid)).null?
+  end
+
+  # Checks to see if group with a given groupname exists.
+  #
+  # Returns: `true` if the group exists.
+  #
+  # ```
+  # Group.exists?(0)     # => true
+  # Group.exists?(32766) # => false
+  # ```
+  def self.exists?(group : String | Symbol) : Bool
+    return !LibC.getgrnam(group.to_s.check_no_null_byte).null?
+  end
+
+  # Returns the group specified by the group ID.
+  # ```
+  # Group[0]
+  # ```
+  def self.[](gid : Int) : Group
+    return new(gid)
+  end
+
+  # Returns the group specified by the groupname.
+  # ```
+  # Group["root"]
+  # ```
+  def self.[](groupname : String | Symbol) : Group
+    return new(groupname)
+  end
+
+  # Initializes a group by the given group ID.
+  def initialize(gid : Int)
+    group_struct = LibC.getgrgid(Group.int_to_gid(gid))
+    raise NotFound.new("Group with gid '#{gid}', was not found.") if group_struct.null?
+
+    group_struct = group_struct.value
+    @name = String.new(group_struct.gr_name)
+    @gid = group_struct.gr_gid
+    @member_names = group_struct.gr_mem.to_slice_null_terminated(LibC::NGROUPS_MAX).to_a() { |chars| String.new(chars) }
+  end
+
+  # Initializes a group by the given groupname.
+  def initialize(groupname : String | Symbol)
+    group_struct = LibC.getgrnam(groupname.to_s.check_no_null_byte)
+    raise NotFound.new("Group with name '#{groupname.to_s}', was not found.") if group_struct.null?
+
+    group_struct = group_struct.value
+    @name = String.new(group_struct.gr_name)
+    @gid = group_struct.gr_gid
+    @member_names = group_struct.gr_mem.to_slice_null_terminated(LibC::NGROUPS_MAX).to_a() { |chars| String.new(chars) }
+  end
+
+  # Returns the groups name.
+  #
+  # ```
+  # Group[0].name # => "root"
+  # ```
+  getter(name : String)
```

Comment:
We typically don't bracket these macros so as to make them appear like keywords. `getter name : String`

Reply:
Fixed